### ISSUE TITLE
Error while running example

### ISSUE THREAD

**Issue by suvigya91** [radical-cybertools/radical.pilot#934]
Hello,
I tried running 00_getting_started.py and had following error. Please help me resolving it.

Thanks

```
 Getting Started (RP version 0.37.10)                                           
--------------------------------------------------------------------------------
gather results 

wait for 128 unit(s)
        -2015-12-21 08:10:58,905: radical.pilot       : MainProcess                     : Thread-1       : ERROR   : [Callback]: ComputePilot 'pilot.0000' failed -- calling exit
2015-12-21 08:10:58,906: radical.pilot       : MainProcess                     : Thread-1       : ERROR   : pilot manager controller thread caught system exit -- forcing application shutdown
Traceback (most recent call last):
  File "/home/suvigya/ve/local/lib/python2.7/site-packages/radical/pilot/controller/pilot_manager_controller.py", line 335, in run
    self.call_callbacks(pilot_id, new_state)
  File "/home/suvigya/ve/local/lib/python2.7/site-packages/radical/pilot/controller/pilot_manager_controller.py", line 237, in call_callbacks
    cb_func(self._shared_data[pilot_id]['facade_object'](), new_state)
  File "/home/suvigya/ve/local/lib/python2.7/site-packages/radical/pilot/pilot_manager.py", line 239, in _default_pilot_error_cb
    sys.exit(1)
SystemExit: 1
exit requested
```

**Comment by vivek-bala** [comment]
Can you run it again with RADICAL_PILOT_VERBOSE set to DEBUG.
```export RADICAL_PILOT_VERBOSE=DEBUG```

**Comment by vivek-bala** [comment]
Also please set RADICAL_PILOT_AGENT_VERBOSE to DEBUG.
```export RADICAL_PILOT_AGENT_VERBOSE=DEBUG```

**Comment by suvigya91** [comment]
Hi,
Here are the logs just before the error. I have also attached the complete log file 
[error.txt](https://github.com/radical-cybertools/radical.pilot/files/69287/error.txt):

2015-12-21 17:47:16,655: radical.pilot       : MainProcess                     : Thread-3       : INFO    : [Callback]: unit unit.000122 state on pilot pilot.0000: AgentStagingInputPending.
2015-12-21 17:47:16,655: radical.pilot       : MainProcess                     : Thread-3       : INFO    : [Callback]: unit unit.000123 state on pilot pilot.0000: AgentStagingInputPending.
2015-12-21 17:47:16,655: radical.pilot       : MainProcess                     : Thread-3       : INFO    : [Callback]: unit unit.000124 state on pilot pilot.0000: AgentStagingInputPending.
2015-12-21 17:47:16,656: radical.pilot       : MainProcess                     : Thread-3       : INFO    : [Callback]: unit unit.000125 state on pilot pilot.0000: AgentStagingInputPending.
2015-12-21 17:47:16,656: radical.pilot       : MainProcess                     : Thread-3       : INFO    : [Callback]: unit unit.000126 state on pilot pilot.0000: AgentStagingInputPending.
2015-12-21 17:47:16,656: radical.pilot       : MainProcess                     : Thread-3       : INFO    : [Callback]: unit unit.000127 state on pilot pilot.0000: AgentStagingInputPending.
2015-12-21 17:47:16,657: radical.pilot       : MainProcess                     : Thread-3       : INFO    : [Callback]: unit unit.000117 state on pilot pilot.0000: AgentStagingInputPending.

2015-12-21 17:47:18,271: radical.pilot       : MainProcess                     : PilotLauncherWorker-1: DEBUG   : Bootstrap command line: /bin/bash ['-l bootstrap_1.sh', " -d 'radical.utils-0.38.tar.gz:saga-python-0.39.tar.gz:radical.pilot-0.37.10.tar.gz' -m 'create' -p 'pilot.0000' -r 'debug' -s 'rp.session.suvigya.suvigya.016790.0023' -v '/home/suvigya/radical.pilot.sandbox/ve_localhost' -a 'multicore'"]
2015-12-21 17:47:18,272: radical.pilot       : MainProcess                     : PilotLauncherWorker-1: DEBUG   : Submitting SAGA job with description: {'Project': None, 'Executable': '/bin/bash', 'TotalPhysicalMemory': None, 'WorkingDirectory': '/home/suvigya/radical.pilot.sandbox/rp.session.suvigya.suvigya.016790.0023-pilot.0000/', 'Queue': None, 'Environment': {}, 'WallTimeLimit': 15, 'Arguments': ['-l bootstrap_1.sh', " -d 'radical.utils-0.38.tar.gz:saga-python-0.39.tar.gz:radical.pilot-0.37.10.tar.gz' -m 'create' -p 'pilot.0000' -r 'debug' -s 'rp.session.suvigya.suvigya.016790.0023' -v '/home/suvigya/radical.pilot.sandbox/ve_localhost' -a 'multicore'"], 'ProcessesPerHost': None, 'Error': 'bootstrap_1.err', 'Output': 'bootstrap_1.out', 'TotalCPUCount': 64}
2015-12-21 17:47:18,492: radical.pilot       : MainProcess                     : PilotLauncherWorker-1: DEBUG   : SAGA job submitted with job id [fork://localhost/]-[6529.0]
2015-12-21 17:47:19,396: radical.pilot       : MainProcess                     : Thread-1       : INFO    : ComputePilot 'pilot.0000' state changed from 'Launching' to 'PendingActive'.
2015-12-21 17:47:19,396: radical.pilot       : MainProcess                     : Thread-1       : INFO    : [Callback]: ComputePilot 'pilot.0000' state: PendingActive.
2015-12-21 17:47:24,411: radical.pilot       : MainProcess                     : Thread-1       : INFO    : ComputePilot 'pilot.0000' state changed from 'PendingActive' to 'Failed'.
2015-12-21 17:47:24,411: radical.pilot       : MainProcess                     : Thread-1       : INFO    : [Callback]: ComputePilot 'pilot.0000' state: Failed.
2015-12-21 17:47:24,411: radical.pilot       : MainProcess                     : Thread-1       : ERROR   : [Callback]: ComputePilot 'pilot.0000' failed -- calling exit
2015-12-21 17:47:24,412: radical.pilot       : MainProcess                     : Thread-1       : ERROR   : pilot manager controller thread caught system exit -- forcing application shutdown
Traceback (most recent call last):
  File "/home/suvigya/ve/local/lib/python2.7/site-packages/radical/pilot/controller/pilot_manager_controller.py", line 335, in run
    self.call_callbacks(pilot_id, new_state)
  File "/home/suvigya/ve/local/lib/python2.7/site-packages/radical/pilot/controller/pilot_manager_controller.py", line 237, in call_callbacks
    cb_func(self._shared_data[pilot_id]['facade_object'](), new_state)
  File "/home/suvigya/ve/local/lib/python2.7/site-packages/radical/pilot/pilot_manager.py", line 239, in _default_pilot_error_cb
    sys.exit(1)
SystemExit: 1
2015-12-21 17:47:24,413: radical.pilot       : MainProcess                     : Thread-1       : DEBUG   : pworker Thread-1 stops   launcher PilotLauncherWorker-1
2015-12-21 17:47:24,413: radical.pilot       : MainProcess                     : Thread-1       : DEBUG   : launcher PilotLauncherWorker-1 stopping
2015-12-21 17:47:24,503: radical.pilot       : MainProcess                     : Thread-1       : DEBUG   : launcher PilotLauncherWorker-1 stopped
2015-12-21 17:47:24,503: radical.pilot       : MainProcess                     : Thread-1       : DEBUG   : pworker Thread-1 stopped launcher PilotLauncherWorker-1
2015-12-21 17:47:24,674: radical.pilot       : MainProcess                     : MainThread     : DEBUG   : session rp.session.suvigya.suvigya.016790.0023 closing
2015-12-21 17:47:24,675: radical.pilot       : MainProcess                     : MainThread     : DEBUG   : session rp.session.suvigya.suvigya.016790.0023 closes   pmgr   pmgr.0000
2015-12-21 17:47:24,675: radical.pilot       : MainProcess                     : MainThread     : DEBUG   : pmgr    pmgr.0000 closing
2015-12-21 17:47:24,675: radical.pilot       : MainProcess                     : MainThread     : DEBUG   : pmgr    pmgr.0000 cancel   launcher Thread-1
2015-12-21 17:47:24,675: radical.pilot       : MainProcess                     : MainThread     : DEBUG   : pworker Thread-1 disables launcher PilotLauncherWorker-1
2015-12-21 17:47:24,675: radical.pilot       : MainProcess                     : MainThread     : DEBUG   : launcher PilotLauncherWorker-1 disabling
2015-12-21 17:47:24,675: radical.pilot       : MainProcess                     : MainThread     : DEBUG   : launcher PilotLauncherWorker-1 disabled
2015-12-21 17:47:24,675: radical.pilot       : MainProcess                     : MainThread     : DEBUG   : pworker Thread-1 disabled launcher PilotLauncherWorker-1
2015-12-21 17:47:24,675: radical.pilot       : MainProcess                     : MainThread     : DEBUG   : pmgr    pmgr.0000 canceled launcher Thread-1
2015-12-21 17:47:24,675: radical.pilot       : MainProcess                     : MainThread     : DEBUG   : Reconnected to existing ComputePilot {'uid': 'pilot.0000', 'stdout': None, 'start_time': None, 'resource_detail': {'cores_per_node': None, 'nodes': None}, 'submission_time': 1450738033.398188, 'logfile': None, 'resource': u'local.localhost', 'log': [<radical.pilot.logentry.Logentry object at 0x7fba6be92b90>, <radical.pilot.logentry.Logentry object at 0x7fba6be92c50>, <radical.pilot.logentry.Logentry object at 0x7fba6be92fd0>, <radical.pilot.logentry.Logentry object at 0x7fba8c1b5290>, <radical.pilot.logentry.Logentry object at 0x7fba8c1dba50>, <radical.pilot.logentry.Logentry object at 0x7fba8c1dbad0>, <radical.pilot.logentry.Logentry object at 0x7fba8c1dba90>, <radical.pilot.logentry.Logentry object at 0x7fba8c16fb50>, <radical.pilot.logentry.Logentry object at 0x7fba8c16fad0>, <radical.pilot.logentry.Logentry object at 0x7fba8c16fb90>, <radical.pilot.logentry.Logentry object at 0x7fba88093090>], 'sandbox': u'file://localhost/home/suvigya/radical.pilot.sandbox/rp.session.suvigya.suvigya.016790.0023-pilot.0000/', 'state': u'Failed', 'stop_time': 1450738043.520044, 'stderr': None}
2015-12-21 17:47:24,675: radical.pilot       : MainProcess                     : MainThread     : DEBUG   : pmgr    pmgr.0000 cancels  pilot  pilot.0000
2015-12-21 17:47:24,677: radical.pilot       : MainProcess                     : MainThread     : INFO    : Sent 'COMMAND_CANCEL_PILOT' command to pilots ['pilot.0000'].
2015-12-21 17:47:24,677: radical.pilot       : MainProcess                     : MainThread     : DEBUG   : can't actively cancel pilot pilot.0000: already in final state
2015-12-21 17:47:24,677: radical.pilot       : MainProcess                     : MainThread     : DEBUG   : pmgr    pmgr.0000 cancel   worker Thread-1
2015-12-21 17:47:24,677: radical.pilot       : MainProcess                     : MainThread     : DEBUG   : pworker Thread-1 stops   launcher PilotLauncherWorker-1
2015-12-21 17:47:24,677: radical.pilot       : MainProcess                     : MainThread     : DEBUG   : launcher PilotLauncherWorker-1 stopping
2015-12-21 17:47:24,677: radical.pilot       : MainProcess                     : MainThread     : DEBUG   : launcher PilotLauncherWorker-1 stopped
2015-12-21 17:47:24,677: radical.pilot       : MainProcess                     : MainThread     : DEBUG   : pworker Thread-1 stopped launcher PilotLauncherWorker-1
2015-12-21 17:47:24,677: radical.pilot       : MainProcess                     : MainThread     : DEBUG   : pmgr    pmgr.0000 canceled worker Thread-1
2015-12-21 17:47:24,677: radical.pilot       : MainProcess                     : MainThread     : DEBUG   : pmgr    pmgr.0000 stops    worker Thread-1
2015-12-21 17:47:24,677: radical.pilot       : MainProcess                     : MainThread     : DEBUG   : pworker Thread-1 stopping
2015-12-21 17:47:24,678: radical.pilot       : MainProcess                     : MainThread     : DEBUG   : pworker Thread-1 stopped
2015-12-21 17:47:24,678: radical.pilot       : MainProcess                     : MainThread     : DEBUG   : pmgr    pmgr.0000 stopped  worker Thread-1
2015-12-21 17:47:24,678: radical.pilot       : MainProcess                     : MainThread     : DEBUG   : pmgr    pmgr.0000 closed
2015-12-21 17:47:24,678: radical.pilot       : MainProcess                     : MainThread     : DEBUG   : session rp.session.suvigya.suvigya.016790.0023 closed   pmgr   pmgr.0000
2015-12-21 17:47:24,678: radical.pilot       : MainProcess                     : MainThread     : DEBUG   : session rp.session.suvigya.suvigya.016790.0023 closes   umgr   umgr.0000
2015-12-21 17:47:24,678: radical.pilot       : MainProcess                     : MainThread     : DEBUG   : uworker  Thread-3 stopping
2015-12-21 17:47:24,982: radical.pilot       : MainProcess                     : Thread-3       : DEBUG   : uworker Thread-3 stops   itransfer InputFileTransferWorker-1
2015-12-21 17:47:24,982: radical.pilot       : MainProcess                     : Thread-3       : DEBUG   : itransfer InputFileTransferWorker-1 stopping
2015-12-21 17:47:25,628: radical.pilot       : MainProcess                     : Thread-3       : DEBUG   : itransfer InputFileTransferWorker-1 stopped
2015-12-21 17:47:25,628: radical.pilot       : MainProcess                     : Thread-3       : DEBUG   : uworker Thread-3 stopped itransfer InputFileTransferWorker-1
2015-12-21 17:47:25,628: radical.pilot       : MainProcess                     : Thread-3       : DEBUG   : uworker Thread-3 stops   itransfer InputFileTransferWorker-2
2015-12-21 17:47:25,629: radical.pilot       : MainProcess                     : Thread-3       : DEBUG   : itransfer InputFileTransferWorker-2 stopping
2015-12-21 17:47:25,629: radical.pilot       : MainProcess                     : Thread-3       : DEBUG   : itransfer InputFileTransferWorker-2 stopped
2015-12-21 17:47:25,629: radical.pilot       : MainProcess                     : Thread-3       : DEBUG   : uworker Thread-3 stopped itransfer InputFileTransferWorker-2
2015-12-21 17:47:25,629: radical.pilot       : MainProcess                     : Thread-3       : DEBUG   : uworker Thread-3 stops   otransfer OutputFileTransferWorker-1
2015-12-21 17:47:25,629: radical.pilot       : MainProcess                     : Thread-3       : DEBUG   : otransfer OutputFileTransferWorker-1 stopping
2015-12-21 17:47:25,629: radical.pilot       : MainProcess                     : Thread-3       : DEBUG   : otransfer OutputFileTransferWorker-1 stopped
2015-12-21 17:47:25,630: radical.pilot       : MainProcess                     : Thread-3       : DEBUG   : uworker Thread-3 stopped otransfer OutputFileTransferWorker-1
2015-12-21 17:47:25,630: radical.pilot       : MainProcess                     : Thread-3       : DEBUG   : uworker Thread-3 stops   otransfer OutputFileTransferWorker-2
2015-12-21 17:47:25,630: radical.pilot       : MainProcess                     : Thread-3       : DEBUG   : otransfer OutputFileTransferWorker-2 stopping
2015-12-21 17:47:25,630: radical.pilot       : MainProcess                     : Thread-3       : DEBUG   : otransfer OutputFileTransferWorker-2 stopped
2015-12-21 17:47:25,630: radical.pilot       : MainProcess                     : Thread-3       : DEBUG   : uworker Thread-3 stopped otransfer OutputFileTransferWorker-2
2015-12-21 17:47:25,630: radical.pilot       : MainProcess                     : MainThread     : DEBUG   : uworker  Thread-3 stopped
2015-12-21 17:47:25,630: radical.pilot       : MainProcess                     : MainThread     : INFO    : Closed UnitManager umgr.0000.
2015-12-21 17:47:25,630: radical.pilot       : MainProcess                     : MainThread     : DEBUG   : session rp.session.suvigya.suvigya.016790.0023 closed   umgr   umgr.0000
2015-12-21 17:47:25,776: radical.pilot       : MainProcess                     : MainThread     : DEBUG   : session rp.session.suvigya.suvigya.016790.0023 closed

**Comment by andre-merzky** [comment]
Thanks for that log!  Could you please also send the content of `/home/suvigya/radical.pilot.sandbox/rp.session.suvigya.suvigya.016790.0023-pilot.0000/bootstrap_1.out` and `~.err`?  Thanks!

**Comment by suvigya91** [comment]
Hi,
Here are the logs for:
bootstrap_1.out: [bootstrap_1_out.txt](https://github.com/radical-cybertools/radical.pilot/files/69387/bootstrap_1_out.txt)

bootstrap_1.err: [bootstrap_1_err.txt](https://github.com/radical-cybertools/radical.pilot/files/69388/bootstrap_1_err.txt)

Thanks!

**Comment by andre-merzky** [comment]
Thanks!  Those look good actually - which is nice, but does not help to find the problem :P  Would you mind to create a tarball of the whole sandbox, and to mail it to me (andre@merzky.net)?

```
tar zcvf rp.session.suvigya.suvigya.016790.0023-pilot.0000.tgz /home/suvigya/radical.pilot.sandbox/rp.session.suvigya.suvigya.016790.0023-pilot.0000/
```

Thanks!

**Comment by suvigya91** [comment]
Hi,
I have emailed you the required tarball.
Thanks!

**Comment by shantenujha** [comment]
Please use gist https://gist.github.com/

**Comment by andre-merzky** [comment]
Thanks - that helped to localize the error to:

```
2015-12-21 17:47:23,517: agent_0.bootstrap_3 : MainProcess                     : MainThread     : ERROR   : Error running agent: agent_0
Traceback (most recent call last):
  File "/home/suvigya/radical.pilot.sandbox/rp.session.suvigya.suvigya.016790.0023-pilot.0000/rp_install/bin/radical-pilot-agent-multicore.py", line 6139, in bootstrap_3
    nodeip = LRMS.hostip(cfg.get('network_interface'))
  File "/home/suvigya/radical.pilot.sandbox/rp.session.suvigya.suvigya.016790.0023-pilot.0000/rp_install/bin/radical-pilot-agent-multicore.py", line 2459, in hostip
    ip = netifaces.ifaddresses(pref)[af][0]['addr']
KeyError: 2
```

Would  you mind sending me the output of `ifconfig -a` from your localhost?  Thanks!

**Comment by suvigya91** [comment]
Hi, 
Here is the output:

          Link encap:Local Loopback  
          inet addr:127.0.0.1  Mask:255.0.0.0
          inet6 addr: ::1/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:17883 errors:0 dropped:0 overruns:0 frame:0
          TX packets:17883 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0 
          RX bytes:3723457 (3.7 MB)  TX bytes:3723457 (3.7 MB)



Thanks!

**Comment by andre-merzky** [comment]
Interesting -- so there is no internet connectivity on that host whatsoever?  That is not a usecase we considered :)  Let me try to reproduce the error with that setting.  Thanks!

**Comment by andre-merzky** [comment]
Hmm, that works all right for me, but the problem does come from the network introspection, according to the logs.

If you don't mind, would you please install radical.pilot from the issue/934 branch, together with saga-python and radical.utils from the devel branches?  If you need help with that, please let me know!  Once installed, please run the example again.  It will likely still fail, but please send `agent_0.bootstrap_3.log` from the pilot sandbox, that should contain mir information then.

Many thanks!  Andre.

**Comment by suvigya91** [comment]
Hi,
I have installed all the 3 things. Here is the log file: 
[agent_0.bootstrap_3_log.txt](https://github.com/radical-cybertools/radical.pilot/files/71440/agent_0.bootstrap_3_log.txt)

I have downloaded project from 3 branches and ran 

python setup.py install 

to install them. Please let me know if I have done something wrong.
Thanks!

**Comment by andre-merzky** [comment]
Could you install the packages via `pip install --upgrade .`, please?  `python setup.py` will unfortunately result in an incomplete installation :/  Thanks!

**Comment by suvigya91** [comment]
Hi Andre,
Sorry for the late reply. Here is the log file: 
[agent_0.bootstrap_3_log.txt](https://github.com/radical-cybertools/radical.pilot/files/73708/agent_0.bootstrap_3_log.txt)

Thanks!

**Comment by marksantcroos** [comment]
And did you try the suggestion about the `pip install --upgrade .` installation?

**Comment by suvigya91** [comment]
Hi,
I tried re-installing it using the above command. I think it did not get installed properly earlier. Here is the new log file: [agent_0.bootstrap_3_log.txt](https://github.com/radical-cybertools/radical.pilot/files/74701/agent_0.bootstrap_3_log.txt)

Thanks!

**Comment by andre-merzky** [comment]
FWIW, Vivek stumbled over the same problem:

```
 23 2016-01-07 17:20:19,066: agent_0.bootstrap_3 : MainProcess                     : MainThread     : DEBUG   : start_bridges done
 24 2016-01-07 17:20:20,072: agent_0.bootstrap_3 : MainProcess                     : MainThread     : ERROR   : Error running agent: agent_0
 25 Traceback (most recent call last):
 26   File "/home/vivek/radical.pilot.sandbox/rp.session.vivek-Lenovo-IdeaPad-Y480.vivek.016807.0013-pilot.0000/rp_install/bin/radical-pilot-agent-multicore.py", line 6139, in bootstrap_3
 27     nodeip = LRMS.hostip(cfg.get('network_interface'))
 28   File "/home/vivek/radical.pilot.sandbox/rp.session.vivek-Lenovo-IdeaPad-Y480.vivek.016807.0013-pilot.0000/rp_install/bin/radical-pilot-agent-multicore.py", line 2459, in hostip
 29     ip = netifaces.ifaddresses(pref)[af][0]['addr']
 30 KeyError: 2
 31 2016-01-07 17:20:20,073: agent_0.bootstrap_3 : MainProcess                     : MainThread     : DEBUG   : agent agent_0 finalized
```

also:

```
$ ifconfig -a
eth0      Link encap:Ethernet  HWaddr b8:88:e3:8e:48:02  
          UP BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)
          Interrupt:16 

lo        Link encap:Local Loopback  
          inet addr:127.0.0.1  Mask:255.0.0.0
          inet6 addr: ::1/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:1361581 errors:0 dropped:0 overruns:0 frame:0
          TX packets:1361581 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0 
          RX bytes:106177856 (106.1 MB)  TX bytes:106177856 (106.1 MB)

wlan0     Link encap:Ethernet  HWaddr 9c:4e:36:7e:f2:a8  
          inet addr:192.168.1.139  Bcast:192.168.1.255  Mask:255.255.255.0
          inet6 addr: fe80::9e4e:36ff:fe7e:f2a8/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:6016597 errors:0 dropped:0 overruns:0 frame:0
          TX packets:1971347 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:7849012829 (7.8 GB)  TX bytes:280845124 (280.8 MB)
```

**Comment by marksantcroos** [comment]
I think this is the same issue as #880.

The solution is probably to also check whether there is an (ipv4) address configured on the interface. Still not fool proof, but good enough likely.

**Comment by marksantcroos** [comment]
I posted instructions in #880. If it is indeed the same issue we can close this one.

**Comment by andre-merzky** [comment]
Yes, looks like the same - closed in favor of #880 then.  Thanks!

**Comment by suvigya91** [comment]
Hi, 
Sorry for the late reply. I followed the instructions from #880 and It works now! 
Thanks!
